x/crosschain/keeper/msg_server_add_to_outtx_tracker.go

```diff
@@ -0,0 +1,181 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"strings"
+
+	cosmoserrors "cosmossdk.io/errors"
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/btcsuite/btcutil"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	eth "github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/zeta-chain/zetacore/common"
+	"github.com/zeta-chain/zetacore/x/crosschain/types"
+	observertypes "github.com/zeta-chain/zetacore/x/observer/types"
+	"github.com/zeta-chain/zetacore/zetaclient/config"
+)
+
+// AddToOutTxTracker adds a new record to the outbound transaction tracker.
+// only the admin policy account and the observer validators are authorized to broadcast this message without proof.
+func (k msgServer) AddToOutTxTracker(goCtx context.Context, msg *types.MsgAddToOutTxTracker) (*types.MsgAddToOutTxTrackerResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	chain := k.zetaObserverKeeper.GetParams(ctx).GetChainFromChainID(msg.ChainId)
+	if chain == nil {
+		return nil, observertypes.ErrSupportedChains
+	}
+
+	if msg.Proof == nil { // without proof, only certain accounts can send this message
+		adminPolicyAccount := k.zetaObserverKeeper.GetParams(ctx).GetAdminPolicyAccount(observertypes.Policy_Type_group1)
+		isAdmin := msg.Creator == adminPolicyAccount
+		isObserver := k.zetaObserverKeeper.IsAuthorized(ctx, msg.Creator, chain)
+
+		// Sender needs to be either the admin policy account or an observer
+		if !(isAdmin || isObserver) {
+			return nil, cosmoserrors.Wrap(observertypes.ErrNotAuthorized, fmt.Sprintf("Creator %s", msg.Creator))
+		}
+	}
+
+	isProven := false
+	if msg.Proof != nil { // verify proof when it is provided
+		txBytes, err := k.VerifyProof(ctx, msg.Proof, msg.ChainId, msg.BlockHash, msg.TxIndex)
+		if err != nil {
+			return nil, types.ErrProofVerificationFail.Wrapf(err.Error())
+		}
+		err = k.VerifyOutTxBody(ctx, msg, txBytes)
+		if err != nil {
+			return nil, types.ErrTxBodyVerificationFail.Wrapf(err.Error())
+		}
+		isProven = true
+	}
+
+	tracker, found := k.GetOutTxTracker(ctx, msg.ChainId, msg.Nonce)
+	hash := types.TxHashList{
+		TxHash:   msg.TxHash,
+		TxSigner: msg.Creator,
+	}
+	if !found {
+		k.SetOutTxTracker(ctx, types.OutTxTracker{
+			Index:    "",
+			ChainId:  chain.ChainId,
+			Nonce:    msg.Nonce,
+			HashList: []*types.TxHashList{&hash},
+		})
+		return &types.MsgAddToOutTxTrackerResponse{}, nil
+	}
+
+	var isDup = false
+	for _, hash := range tracker.HashList {
+		if strings.EqualFold(hash.TxHash, msg.TxHash) {
+			isDup = true
+			if isProven {
+				hash.Proved = true
```

Comment:
Does this value get modified in the tracker? If not then the variable has no use here.

Reply:
The `hash.Proved` value is used nowhere for now. Moving forward, we'll gradually utilize the `Proved` flag in a way that whenever a tracker hash gets its `Proved` flag set to `true`, it will be treated as a true hash of outTx/inTx and be trusted by any party.